tests/test_mailchimp_bookmarks.py

```diff
@@ -0,0 +1,211 @@
+                        self.BOOKMARK_DATETIME_FORMAT,
+                    )
+
+                    # Verify the first sync sets a bookmark of the expected form
+                    self.assertIsNotNone(first_bookmark_value)
+
+                    # Verify the second sync sets a bookmark of the expected form
+                    self.assertIsNotNone(second_bookmark_value)
+
+                    # Verify the second sync bookmark is Equal to the first sync bookmark
+                    self.assertEqual(second_bookmark_value, first_bookmark_value)
+
+                    for record in first_sync_messages:
+
+                        # Verify the first sync bookmark value is the max replication key value for a given stream
+                        replication_key_value = self.parse_date(
+                            record.get(replication_key), self.BOOKMARK_DATETIME_FORMAT
+                        )
+
+                        self.assertLessEqual(
+                            replication_key_value,
+                            first_bookmark_value_ts,
+                            msg="First sync bookmark was set incorrectly, a record with a greater replication-key value was synced.",
+                        )
+
+                    for record in second_sync_messages:
+                        # Verify the second sync replication key value is Greater or Equal to the first sync bookmark
+                        replication_key_value = self.parse_date(
+                            record.get(replication_key), self.BOOKMARK_DATETIME_FORMAT
+                        )
+                        self.assertGreaterEqual(
+                            replication_key_value,
+                            simulated_bookmark_value,
+                            msg="Second sync records do not respect the previous bookmark.",
+                        )
+
+                        # Verify the second sync bookmark value is the max replication key value for a given stream
+                        self.assertLessEqual(
+                            replication_key_value,
+                            second_bookmark_value_ts,
+                            msg="Second sync bookmark was set incorrectly, a record with a greater replication-key value was synced.",
+                        )
+                
+                        # Verify that you get less data the 2nd time around
+                        self.assertLess(
+                            second_sync_count,
+                            first_sync_count,
+                            msg="second sync didn't have less records, bookmark usage not verified",
+                        )
+
+                        # Verify at least 1 record was replicated in the second sync
+                        self.assertGreater(
+                            second_sync_count,
+                            0,
+                            msg=f"We are not fully testing bookmarking for {stream}",
+                        )
```

Comment:
This assertion is already written on L#107-L#112, please remove it.

Reply:
removed duplicate assertions